build/commands/lib/util.js

```diff
@@ -252,6 +252,24 @@ const util = {
     fileMap.add([path.join(config.braveCoreDir, 'ui', 'webui', 'resources', 'css', 'text_defaults_md.css'),
                  path.join(config.srcDir, 'ui', 'webui', 'resources', 'css', 'text_defaults_md.css')])
 
+    let explicitSourceFiles = new Set()
+    if (process.platform === 'darwin') {
+      // Set proper mac app icon for channel to chrome/app/theme/mac/app.icns.
+      // Each channel's app icons are stored in brave/app/theme/$channel/app.icns.
+      // With this copying, we don't need to modify chrome/BUILD.gn for this.
+      const iconSource = path.join(braveAppDir, 'theme', 'brave', 'mac', config.channel, 'app.icns')
```

Comment:
Why aren't we just copying all channels here?

Reply:
source file depends on the channel, but the destination file is not.